### PR TITLE
[#1951] Handle script based DReps

### PR DESCRIPTION
**Pull request by jdyczka**
## List of changes

- Handle script based DReps

## Checklist

- [related issue](https://github.com/IntersectMBO/govtool/issues/1951)
- [x] My changes generate no new warnings
- [x] My code follows the [style guidelines](https://github.com/IntersectMBO/govtool/tree/main/docs/style-guides) of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the [changelog](https://github.com/IntersectMBO/govtool/blob/main/CHANGELOG.md)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: govtool/backend/src/VVA/API/Types.hs [hunk 1]
```diff
@@ -758,7 +759,8 @@ instance ToSchema DRepType where
 
 data DRep
   = DRep
-      { dRepDrepId                 :: DRepHash
+      { dRepIsScriptBased          :: Bool
+      , dRepDrepId                 :: DRepHash
```
Comment (Ryun1) [inline govtool/backend/src/VVA/API/Types.hs:763]:
```suggestion
      , dRepID                 :: DRepHash
```

Comment (MSzalowski) [reply govtool/backend/src/VVA/API/Types.hs:763]: We are returning `drepId` not just `id`

File: govtool/frontend/src/context/wallet.tsx [hunk 2]
```diff
@@ -657,8 +657,10 @@ const CardanoProvider = (props: Props) => {
           targetDRep = DRep.new_always_abstain();
         } else if (target === AutomatedVotingOptionDelegationId.no_confidence) {
           targetDRep = DRep.new_always_no_confidence();
-        } else if (target.includes("drep")) {
+        } else if (target.includes("drep1")) {
           targetDRep = DRep.new_key_hash(Ed25519KeyHash.from_bech32(target));
+        } else if (target.includes("drep_script1")) {
+          targetDRep = DRep.new_script_hash(Ed25519KeyHash.from_bech32(target));
```
Comment (MSzalowski) [inline govtool/frontend/src/context/wallet.tsx:663]: target should be:
```suggestion
          targetDRep = DRep.new_script_hash(ScriptHash.from_hex(target));
```

where `target` is the `scriptHash` of the script based DRep that is missing in the SQL.

I'll make the reference implementation of this and share you soon

**Review by MSzalowski — CHANGES_REQUESTED** [review]
If that is easier to depend on the `drep_hash.has_script` that is ok, but still for the script based Drep delegation certificate we need a script hash for such DRep, meaning, that we need to take this value.

So, having the `scriptHash` in a DRep should be just enough to establish that such DRep is script based

**Comment by MSzalowski** [comment]
> If that is easier to depend on the `drep_hash.has_script` that is ok, but still for the script based Drep delegation certificate we need a script hash for such DRep, meaning, that we need to take this value.
> 
> So, having the `scriptHash` in a DRep should be just enough to establish that such DRep is script based

I figured out that dRepId in our backend is already the scriptHash, so there is no need to update the SQL query - all good, just update pls that small change in wallet.tsx :saluting_face:

**Review by MSzalowski — APPROVED** [review]

**Pull request merged**